core/domain/draft_upgrade_services.py

```diff
@@ -134,89 +134,107 @@ def _convert_html_in_draft_change_list(
+                # we used cast here.
+                interaction_cust_arg = cast(
+                    Dict[str, Dict[str, Any]], new_value
+                )
+                if 'choices' in interaction_cust_arg.keys():
                     for value_index, value in enumerate(
-                            new_value['choices']['value']):
+                            interaction_cust_arg['choices']['value']):
                         if isinstance(value, dict) and 'html' in value:
-                            new_value['choices']['value'][value_index][
-                                'html'
-                            ] = conversion_fn(value['html'])
+                            interaction_cust_arg['choices']['value'][
+                                value_index
+                            ]['html'] = conversion_fn(value['html'])
                         elif isinstance(value, str):
-                            new_value['choices']['value'][value_index] = (
-                                conversion_fn(value))
-            elif (change.property_name ==
+                            interaction_cust_arg['choices']['value'][
+                                value_index
+                            ] = conversion_fn(value)
+            elif (edit_state_property_cmd.property_name ==
                   exp_domain.STATE_PROPERTY_WRITTEN_TRANSLATIONS):
-                # Ruling out the possibility of any other type for MyPy
-                # type checking, because here 'new_value' is a value of
-                # attribute that was dynamically defined on change_domain
-                # and that attribute has str type. Thus to convert the
-                # type, we used assert here.
-                assert isinstance(new_value, dict)
+                # Here we use cast because in this 'elif clause' we are
+                # updating the written_translations of a State which can
+                # only be of type WrittenTranslationsDict. So, to rule out
+                # all other property types for MyPy type checking, we used
+                # cast here.
+                written_translations_dict = cast(
+                    state_domain.WrittenTranslationsDict, new_value
+                )
                 for content_id, language_code_to_written_translation in (
-                        new_value['translations_mapping'].items()):
+                        written_translations_dict[
+                            'translations_mapping'
+                        ].items()
+                    ):
                     for language_code in (
                             language_code_to_written_translation.keys()):
-                        new_value['translations_mapping'][
-                            content_id][language_code]['html'] = (
-                                conversion_fn(new_value[
+                        # In _convert_* functions, we allow less strict typing
```

Comment:
Confirming this rationale is fine in general, but please explain which specific fields are discontinued in this particular case (it's not clear).

Reply:
Here 'html' field is missing from the latest domain object. This is also updated in the code comment. Here `new_value['translations_mapping'][content_id][language_code]` is of type `WrittenTranslationDict` and `WrittenTranslationDict` don not contain HTML field.

https://github.com/oppia/oppia/blob/37ff4591fc7e57f273e54ac5a0226b8561086dbf/core/domain/state_domain.py#L1923

Reply:
Um. In that case, it sounds like this code needs to be updated. Can we have it use the correct field instead? It looks like the current code is just wrong.

/cc @kevintab95

Reply:
Because looking at the code it is not clear to me which field should i replace here.

Reply:
Some context in case it is useful: Edits to written translations are no longer recorded as an edit to a "state property". It has its own commit cmd that is used in the translation editor as well as the contributor dashboard translation suggestion flow (see "exp_domain.CMD_ADD_WRITTEN_TRANSLATION".)

We maintained support for existing suggestion / changelists on the server, see the apply_changelist method here:
https://github.com/oppia/oppia/blob/d4b06399ad2859df6917c5e18ce9024e452351d0/core/domain/exp_services.py#L637-L642

For this specific case, could we add a check to see if "data_format" matches HTML and then run the HTML conversion function?

Reply:
> could we add a check to see if "data_format" matches HTML and only then run the HTML conversion function?

Can please explain this change a bit more @kevintab95 !

Reply:
Sorry if that wasn't clear, let me try again.

The content of written translations can be of different types e.g. HTML, unicode, set of strings etc. (see state_domain.WrittenTranslation) -- "data_format" field stores this info. In this code, we seem to be assuming that there will always be a 'html' key in the dict, this may not be true -- it may be that in the previous schema prior to introduction of data_format, we used the html key (see [comment](https://github.com/oppia/oppia/blob/d65bea24b446331803bd06faed2772905e61c535/core/domain/state_domain.py#L2369-L2373)) but we no longer use it. So we could check if there is a 'html' key and only then update it.

Reply:
Thanks for the clear explanation -- As suggested I have added a check for 'html' key. PTAL!